test/unit/node_shared_test.rb

```diff
@@ -20,6 +20,32 @@ class NodeSharedTest < ActiveSupport::TestCase
     assert_equal 5, html.scan('notes-grid-test').length
     assert html.scan('<td class="title">').length > 1
   end
+ 
+  test 'that NodeShared can be used to convert short codes like [button:foo:https://google.com] into tables which list buttons' do
+    before = "Here are some notes in a table: \n\n[button:Press me:/questions] \n\n[button:Cancel:https://google.com]\n\n`[button:Cancel:https://google.com]` This shouldn't get recognized because it's in ` ticks.\n\nMake sense?"
+    html = NodeShared.button(before)
+    assert html
+    assert_equal 1, html.scan('<table class="table inline-grid notes-grid notes-grid-test notes-grid-test-').length
+    assert_equal 1, html.scan('<table').length
+    assert_equal 5, html.scan('button-grid-test').length
+    assert html.scan('<td class="title">').length > 1
+  end
+
+  test 'that NodeShared does not convert short codes like [button:foo:https://google.com] into tables which list notes, when inside `` marks' do
+    before = "This shouldn't actually produce a table:\n\n`[button:Cancel:https://google.com]`\n\nOr this:\n\n `[button:Cancel:https://google.com]`"
+    html = NodeShared.button(before)
+    assert_equal 0, html.scan('<table class="table inline-grid button-grid button-grid').length
+    assert_equal 0, html.scan('<table').length
+    assert_equal 0, html.scan('button-grid').length
+  end
+
+  test 'that NodeShared does not convert short codes like [button:foo:https://google.com] into tables which list notes, when in code tags' do
+    before = "This shouldn't actually produce a table:\n\n<code>[button:Cancel:https://google.com]</code>"
+    html = NodeShared.button(before)
+    assert_equal 0, html.scan('<table class="table inline-grid button-grid button-grid').length
```

Comment:
Same here for `btn btn-primary` - that way we can be sure your converter did /not/ run when it's inside `<code>` tags!